fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1384,4 +1386,242 @@ private SqlQueryData processMissingParm(Class<?> resourceType, QueryParameter qu
+     *       AND CP1.CODE_SYSTEM_ID = 20004
+     *   )
+     * </pre>
+     *
+     * @see https://www.hl7.org/fhir/search.html#has
+     * @param resourceType
+     *                  - The resource type being searched.
+     * @param queryParm
+     *                  - A Parameter representing a reverse chained query.
+     * @return SqlQueryData
+     *                  - The query segment for a reverse chained parameter reference search.
+     * @throws Exception
+     */
+    @Override
+    protected SqlQueryData processReverseChainedReferenceParm(Class<?> resourceType, QueryParameter queryParm) throws Exception {
+        final String METHODNAME = "processReverseChainedReferenceParm";
+        log.entering(CLASSNAME, METHODNAME, queryParm.toString());
+
+        final String CR = "CR";
+        final String CLR = "CLR";
+        final String CP = "CP";
+        String prevChainedResourceVar = null;
+        String prevChainedLogicalResourceVar = null;
+        String chainedResourceVar = null;
+        String chainedLogicalResourceVar = null;
+        String chainedParmVar = null;
+        String nextChainedResourceVar = null;
+        String nextChainedLogicalResourceVar = null;
+        String nextChainedParmVar = null;
+        QueryParameter previousParm = null;;
+        int parmIndex = 0;
+        int lastParmIndex = queryParm.getChain().size();
+        boolean chainedParmProcessed = false;
+        StringBuilder selectSegments = new StringBuilder();
+        StringBuilder whereClauseSegments = new StringBuilder();
+        List<Object> bindVariables = new ArrayList<>();
+
+        // Loop through the chained query parameters in order
+        List<QueryParameter> queryParms = queryParm.getChain();
+        queryParms.add(0, queryParm);
+        for (QueryParameter currentParm : queryParms) {
+
+            prevChainedResourceVar        = CR + parmIndex;
+            prevChainedLogicalResourceVar = CLR + parmIndex;
+            chainedResourceVar        = CR + (parmIndex + 1);
+            chainedLogicalResourceVar = CLR + (parmIndex + 1);
+            chainedParmVar            = CP + (parmIndex + 1);
+            nextChainedResourceVar        = CR + (parmIndex + 2);
+            nextChainedLogicalResourceVar = CLR + (parmIndex + 2);
+            nextChainedParmVar        = CP + (parmIndex + 2);
+            StringBuilder whereClauseSegment = new StringBuilder();
+
+            if (parmIndex == 0) {
+                // BUild outer select: SELECT CLR0.LOGICAL_ID
+                //                       FROM <resource-type>_LOGICAL_RESOURCES AS CLR0
+                //                       JOIN <resource-type>_RESOURCES AS CR0
+                //                         ON CR0.RESOURCE_ID = CLR0.CURRENT_RESOURCE_ID AND CR0.IS_DELETED = 'N'
+                //                       WHERE
+                selectSegments.append("SELECT " + prevChainedLogicalResourceVar + DOT + "LOGICAL_ID")
```

Comment:
let's break these into individual appends
Also SELECT and I think LogicalId are already in JDBCConstants

Reply:
broke into individual appends, also added several constants to JDBCConstants that were not there, and replaced string literals in this class with the new constants